auth/realm/base/src/main/java/org/wildfly/security/auth/realm/LegacyPropertiesSecurityRealm.java

```diff
@@ -431,6 +431,7 @@ public Builder setUsersStream(InputStream usersStream) {
          * @return this {@link Builder}
          */
         public Builder setGroupsStream(InputStream groupsStream) {
+            Assert.checkNotNullParam("groupsStream", groupsStream);
```

Comment:
Just a minor, I think null is valid value for the attribute groupsStream.  If you take a look here: https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/realm/base/src/main/java/org/wildfly/security/auth/realm/LegacyPropertiesSecurityRealm.java#L289 we do expect this attribute could have been set to null. So it is a valid value for this attribute.

Reply:
I thought this was a redundant check to avoid a null pointer in the next line. Is null really a valid value for groupStream? Removing my null check as we already have a check here

Reply:
Thanks @vijay-boddupalli. This `Builder` class is used to create instances of `LegacyPropertiesSecurityRealm`. In `LegacyPropertiesSecurityRealm`, on the line Diana mentioned above, we can see that `null` is a valid value for `groupsStream` since there is a check for null there. So we don't need the check in `setGroupsStream`.

Reply:
Sounds good, thanks Diana and Farah!